core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -466,6 +467,7 @@ private void cleanUpOnCommitFailure() {
         .run(ops.io()::deleteFile);
   }
 
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
```

Comment:
Rather than supressing this, it's a good sign that the new code should be refactored.